### PR TITLE
*: send socat and wait for sticky disk load

### PR DESCRIPTION
**Pull request by adityamaru**
This change teaches our action to send a socat message to the agent to load the sticky disk. It then waits for an MMDS signal to indicate the load is complete before proceding. If either of these steps fails then sticky disks will not be enabled for that VM.

### CODE REVIEW

**Comment by adityamaru** [comment]
Subject to testing

**Comment by adityamaru** [comment]
tested, this seems to be working now

File: src/main.ts [hunk 1]
```diff
@@ -11,18 +11,91 @@ import { promisify } from 'util';
...
 
 
 import * as context from './context';
 import * as stateHelper from './state-helper';
+import { get } from 'http';
 
 const supportedDockerDriver = 'remote';
 const mountPoint = '/var/lib/buildkit';
 const device = '/dev/vdb';
+const mmdsIPv4Addr = "169.254.169.254";
```
Comment (aayushshah15) [inline src/main.ts:24]: is this supposed to be hardcoded?

File: src/main.ts [hunk 2]
```diff
@@ -11,18 +11,91 @@ import { promisify } from 'util';
...
+      throw new Error('Maximum number of retries exceeded');
+    }
+    try {
+      const result = await command();
+      if (result && !result.includes("Resource not found")) {
+        core.debug(`Command result: ${result}`);
+        return result;
+      }
+    } catch (error) {
+      core.debug(`Command failed: ${error}`);
```
Comment (aayushshah15) [inline src/main.ts:92]: nit: lets change this to `core.warning()`

**Review by aayushshah15 — APPROVED** [review]

File: src/main.ts (hunk 1 above)
Comment (adityamaru) [reply src/main.ts:24]: yeah this is the FC specific IP where MMDS is shared

File: src/main.ts (hunk 2 above)
Comment (adityamaru) [reply src/main.ts:92]: core.warning shows up in the users logs, this retry is likely going to retry atleast once with a 404 until the MMDS is set by the agent. I wanted to completely hide the output from the user

**Pull request merged**